src/Microsoft.AspNet.Mvc.Core/ModelBinding/FormCollectionModelBinder.cs

```diff
@@ -30,19 +29,11 @@ public class FormCollectionModelBinder : IModelBinder
             if (request.HasFormContentType)
             {
                 var form = await request.ReadFormAsync();
-                if (bindingContext.ModelType.GetTypeInfo().IsAssignableFrom(form.GetType().GetTypeInfo()))
```

Comment:
Curious: Why not do this instead of the type equality check at [line 22](https://github.com/aspnet/Mvc/pull/3021/files#diff-9ab5b561d10929ae1a601c3f090a8e03R22)?